packages/core/src/ember-html.ts

```diff
@@ -25,57 +26,84 @@ export interface EmberHTML {
   implicitTestStyles?: Node;
 }
 
-class NodeRange {
```

Comment:
I think there are multiple bugs in the original implementation. For example, we never updated `this.end` when we insert something, so the clear code probably does not work as intended.